dietpi/dietpi-software

```diff
@@ -8568,6 +8585,222 @@ _EOF_
+
+				fi
+
+			done
+
+			local orport=443
+			invalid_text=
+			while :
+			do
+
+				G_WHIP_DEFAULT_ITEM=$orport
+				if G_WHIP_INPUTBOX "${invalid_text}Please enter the ORPort that people will use to connect to your TOR relay:\n
+NB: This port needs to be forwarded by your router and/or opened in your firewall settings. Default value is: 443. Do not use default value if running a webserver\n.
+Do not use 9001, as this port is commonly associated with TOR and websites might scan for it." && disable_error=1 G_CHECK_VALIDINT "$G_WHIP_RETURNED_VALUE" 0; then
+					orport=$G_WHIP_RETURNED_VALUE
+					break
+				else
+					invalid_text='[ERROR] No valid entry found, value needs to be a sequence of integers. Please retry...\n\n'
+				fi
+
+			done
+
+			local choice_required=
+			while :
+			do
+
+				G_WHIP_MENU_ARRAY=(
+
+					'Bridge' ': Use this machine as an obfs4 bridge. This is the safest to run from home, as no information will be publically published.'
+					'Guard/Middle' ': Use this machine as a beginning or middle relay. Your IP address, contact email, and nickname will be made public, and this IP address might be blacklisted by certain services.'
+					'Exit' ': Use this machine as an exit relay. This is most at risk for legal complaints, as it is the node that actually contacts the websites. Everything stated in Bridge/Middle will still apply.'
+
+				)
+
+					G_WHIP_MENU "${choice_required}Please choose which type of relay this should be set up as:" && break
+					choice_required='[ERROR] A choice is required to finish the TOR
+					 install.\n\n'
+			done
+
+			if [[ $G_WHIP_RETURNED_VALUE == 'Bridge' ]]; then
+
+				G_AGI obfs4proxy
+
+				while :
+				do
+
+					local obfs4port=80
+					G_WHIP_DEFAULT_ITEM=$obfs4port
+					if G_WHIP_INPUTBOX "${invalid_text}Please enter the obfs4 port that people will use to connect to your bridge:\n
+	NB: This port needs to be forwarded by your router and/or opened in your firewall settings. Default value is: 80. Do not use default value if running a webserver\n.
+	Do not use 9001, as this port is commonly associated with TOR and websites might scan for it." && disable_error=1 G_CHECK_VALIDINT "$G_WHIP_RETURNED_VALUE" 0; then
+						orport=$G_WHIP_RETURNED_VALUE
+						break
+					else
+						invalid_text='[ERROR] No valid entry found, value needs to be a sequence of integers. Please retry...\n\n'
+					fi
+
+				done
+
+				if (( $obfs4port < 1024 )); then
+					G_EXEC setcap cap_net_bind_service=+ep "$(command -v obfs4proxy)"
+				fi
+
+				cat << _EOF_ > /etc/tor/torrc
+BridgeRelay 1
+
+ORPort $orport
+
+ServerTransportPlugin obfs4 exec $(command -v obfs4proxy)
+
+ServerTransportListenAddr obfs4 0.0.0.0:$obfs4port
+
+ExtORPort auto
+_EOF_
+
+			G_CONFIG_INJECT 'NoNewPrivileges=' 'NoNewPrivileges=no' /lib/systemd/system/tor@default.service
+			G_CONFIG_INJECT 'NoNewPrivileges=' 'NoNewPrivileges=no' /lib/systemd/system/tor@.service
```

Comment:
I'm fans of not editing the packages shipped service files directly, but instead place drop-in configs:
```
G_EXEC mkdir -p /etc/systemd/system/tor@.service.d
G_EXEC eval "echo -e '[Service]\nNoNewPrivileges=no' > /etc/systemd/system/tor@.service.d/dietpi-relay.conf"
```
I'm just not sure if this works for instantiated services, requires a quick test. If it works, `tor@default.service` does not need to be changed as it inherits everything from `tor@.service`.